doc/source/advanced.rst

```diff
@@ -423,33 +423,75 @@ To get information about the current available resource capacity of your cluster
+                "pip": ["chess"]
+            }]
+        }
+    }
 
-    result = ray.get(my_task.options(runtime_env={"conda_env": "my_env"}).remote())
+Different ways to use runtime environments:
 
 .. code-block:: python
 
-    my_actor = MyActor.options(runtime_env={"conda_env": "my_env"}).remote()
+    ray.init(job_config=ray.job_config.JobConfig(runtime_env=runtime_env))
+    ray.client("localhost:10001").env(runtime_env).connect()
+    f.options(runtime_env=runtime_env).remote()
+    Actor.options(runtime_env=runtime_env).remote()
 
-Nested tasks and actors will inherit the conda environment of their parent by default.
 
-To have Ray start all tasks and actors in a specific conda environment by default, you may
-pass in the desired conda environment name into ``ray.init()``:
+The ``runtime_env`` is a dictionary including one or more of the following arguments:
 
-.. code-block:: python
+- ``working_dir`` (Path): Specifies the working directory for your job. This can either be a local directory or zip file.
+  It will be cached on the cluster, so the next time you connect you will be able to skip uploading the directory.
+  Furthermore, if you locally make a small change to your directory, the next time you connect only the updated part will be uploaded.
+
+  - Examples
+
+    - ``"."  # cwd``
+
+    - ``"local_project.zip"  # Archive is unpacked into the directory.``
+
+  Note: Setting this option per-task or per-actor is currently unsupported.
+
+- ``pip`` (List[str] | str): Either a list of pip packages, or a string containing the path to a pip “requirements.txt” file. This will be dynamically installed in the ``runtime_env``.
+
+  - Example: ``["requests==1.0.0", "aiohttp"]``
+
+  - Example: ``"./requirements.txt"``
+
+- ``conda`` (dict | str): Either a dict representing the conda environment YAML, a string containing the path to a 
+  `conda “environment.yaml” <https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually>`_ file,
+  or the name of a local conda env already installed on each node in your cluster (e.g., ``"pytorch_p36"``).
+  In the first two cases, the Ray and Python dependencies will be automatically injected into the environment to ensure compatibility, so there is no need to manually include them.  
+  Note that the ``conda`` and ``pip`` keys of ``runtime_env`` cannot both be specified at the same time--to use them together, please add your conda dependencies under the ``"pip"`` entry in your conda ``environment.yaml``.
+
+  - Example: ``{"conda": {"dependencies": ["pytorch", “torchvision”, "pip", {"pip": ["chess"]}]}}``
+
+  - Example: ``"./environment.yml"``
+
+  - Example: ``"pytorch_p36"``
+
+- ``env_vars`` (Dict[str, str]): Environment variables to set.
+
+  - Example: ``{"OMP_NUM_THREADS": "32", "TF_WARNINGS": "none"}``
+
+The runtime env is inheritable, so it will apply to all tasks/actors within a job and all child tasks/actors of a task or actor, once set.
```

Comment:
I feel like all of this should just be coming from the docstring of runtime_env -- is there a way we can just do the documentation for the individual options there and embed it here?

Reply:
I agree this makes sense as a long term solution, but right now runtime_env is a simple Python dict, so it doesn't have a docstring.  The RuntimeEnvDict class in the code has a docstring, but it's an internal object so I don't think we want to expose it.  I looked for a way for Sphinx to only grab the docstring of RuntimeEnvDict without having the class name appearing in the docs, but couldn't find one.